src/ripple/app/main/Application.cpp

```diff
@@ -1110,7 +1064,30 @@ class ApplicationImp : public Application, public RootStoppable, public BasicApp
                 return validators().trustedPublisher(pubKey);
             });
 
-        stopped();
+        // The order of these stop calls is delicate.
+        // Re-ordering them risks undefined behavior.
+        m_shaMapStore->stop();
+        m_jobQueue->stop();
+        if (shardArchiveHandler_)
+            shardArchiveHandler_->stop();
+        m_nodeStore->stop();
+        if (shardStore_)
+            shardStore_->stop();
+        if (config_->reporting())
+        {
+#ifdef RIPPLED_REPORTING
+            pgPool_->stop();
+#endif
+            reportingETL_->stop();
```

Comment:
I will trust your judgment on it and make that change. Because it is so difficult for me to test with reporting enabled, can I get you to do that for me?

Reply:
Yeah I'll test and report back

Reply:
Actually @thejohnfreeman what was the stopping order of these before this PR? To stay in the spirit of the rest of the PR, I'd prefer to preserve the stopping order actually.

Reply:
That's difficult to answer because (a) do you want to consider the order objects call `stopped()` (which can be non-deterministic) or (b) do you want to consider the order that `RootStoppable::stop` blocked on those calls or (c) do you want to consider the order that `Stoppable`s first learned that they needed to stop (which can be non-deterministic)?